cabal-install/src/Distribution/Client/Errors.hs

```diff
@@ -1,6 +1,9 @@
 {-# LANGUAGE ConstraintKinds #-}
+{-# LANGUAGE DeriveGeneric #-}
 {-# LANGUAGE FlexibleInstances #-}
 {-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# OPTIONS_GHC -fno-warn-incomplete-uni-patterns #-}
```

Comment:
This looks a bit suspect, why is it needed?

Reply:
This is part of existing code (TODO in module `TargetSelector.hs`). I have moved the existing logic to `Errors.hs`, I did not want to make changes to existing code / logic, so I left it unchanged.